responses/tests/test_matchers.py

```diff
@@ -782,3 +783,208 @@ def test_matchers_create_key_val_str():
         "a, {3: test, key1: val1, key2: 2}], test: val}"
     )
     assert conv_str == reference
+
+
+def test_request_matches_headers_regex():
```

Comment:
can you please create a class for headers matcher and wrap everything into a class. That will allow to logically separate tests